server/controller/chore.js

```diff
@@ -1,53 +1,62 @@
-
 import { check, validationResult } from "express-validator";
-
 // MODELS
 import Chore from "../models/Chore.js";
 import User from "../models/User.js";
 
-
-
-
 /*
  ** This function creates new chore
  ** @param {Object} req - The request object
  ** @param {Object} res - The response object
  */
- export const createChore = async (req, res) => {
-    // Sanitize data
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-        return res.status(400).json({ errors: errors.array() });
-      }
+export const createChore = async (req, res) => {
+  // Sanitize data
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
 
-    try {
-        const { name, frequency, location, duration, preference } = req.body; //extracting fields recieved from request
-        // console.log(req.body);
-        console.log(`Task Name: ${name},\nFrequency Quantity & Interval: ${frequency.quantity} times every ${frequency.interval},\nLocation: ${location},\nDuration (s): ${duration},\nPreference: ${preference}`);
+  try {
+    const { name, frequency, location, duration, preference } = req.body; //extracting fields recieved from request
+    // console.log(req.body);
+    console.log(
+      `Task Name: ${name},\nFrequency Quantity & Interval: ${frequency.quantity} times every ${frequency.interval},\nLocation: ${location},\nDuration (s): ${duration},\nPreference: ${preference}`
+    );
 
-        // check if this chore already exists
-        const existingChore = await Chore.findOne({ name });
-        if (existingChore) {
-            return res.status(409).json({ message: `Chore: ${name} already exists. If you want to change somthing, please use the EDIT option.` });
-        }
+    // check if this chore already exists
+    const existingChore = await Chore.findOne({ name });
```

Comment:
I think this might cause some issues, because technically the name of a chore is not unique. The combination of all traits (or just the ID) are the only ways we can identify a chore. For example, the chore could be called "Sweeping", but location could be "Kitchen" or "Garage", or they could do sweeping for 10 minutes on Fridays, but 5 minutes every other day of the week...

Reply:
Thanks for spotting that, I've made the changes